ArduCopter/mode_guided.cpp

```diff
@@ -768,6 +779,163 @@ bool ModeGuided::limit_check()
     return false;
 }
 
+//loiter turns(circle) mode
+
+void ModeGuided::do_circle(const mavlink_command_long_t &packet)
+{
+    const Location circle_center = loc_from_param(packet);
+
+    // calculate radius
+    uint8_t circle_radius_m = radius_from_param(packet); // circle radius held in high byte of p1
+
+    //store the number of turns
+    turn_count = (uint8_t)packet.param1;
+
+    // move to edge of circle (verify_circle) will ensure we begin circling once we reach the edge
+    circle_movetoedge_start(circle_center, circle_radius_m);
+}
+
+Location ModeGuided::loc_from_param(const mavlink_command_long_t &packet)
```

Comment:
Don't duplicate code from Auto, move this into Mode Parent class so we keep only one implementation

Reply:
I understand Sir. I have refactored the duplicated code from Auto that was being used in the Guided (specifically the circle_start() function).I would like to further refactor the code if further changes are required

Reply:
@khancyr 's point here is that this code doesn't really belong here.

I like this function, but it should be a static function on GCS_MAVLINK, something like:
`bool GCS_MAVLINK::location_from_command(Location &loc, const mavlink_command_long_t &packet)`

There's an existing `command_long_stores_location` which should be re-used in that function.